llm4pddl/utils.py

```diff
@@ -238,6 +241,75 @@ def minify_pddl_problem(problem: str) -> str:
     return new_problem
 
 
+def embed_training_tasks(training_tasks: Sequence[Task]) -> List:
+    """"embeds a list of tasks.
+
+    Returns a list of embeddings with indicies corresponding to its
+    task.
+    """
+    embedding_model = SentenceTransformer(FLAGS.embedding_model_name)
+    embeddings = [embed_task(task, embedding_model) for task in training_tasks]
+    return embeddings
+
+
+def embed_task(task: Task, embedding_model: SentenceTransformer) -> List:
+    """Embeds a task using embedding_model.
+
+    Returns a numpy array?
+    """
+    with open(task.problem_file, 'r', encoding='utf-8') as f:
+        task_string = f.read()
+    task_string = minify_pddl_problem(task_string)
```

Comment:
we will want to make sure that this task string matches the task strings that we construct inside of the approach. to make sure of that, let's make a new utility function that takes the task and performs the string construction, and then use that utility function inside of both the approach and here. the new utility function should cut out code from the LLMOpenLoopApproach.

Reply:
Just so I understand this correctly: this new function would do all the construction of the prompt that we have in OpenLoopApproach right now? would that essentially mean taking the chunks of the code in `_create_prompt` that dont need the env and implementing them in utils instead of openloopapproach?

Reply:
right! an alternative would be to move this `embed_task` utility inside `LLMOpenLoopApproach` so that you can use the prompt creation code without moving it into utilities. I don't know if that would be any easier though.

Reply:
wait im trying to fix this right now...wouldn't it be much easier to make `_create_prompt` a public method instead of a private method? Then I could simply call the function and wouldnt have to deal with moving test cases and making sure the functionality is maintained?

Reply:
are you worried about calling `_create_prompt` from within a test? or where would the private-turned-public method be called?

Reply:
wait actually never mind im just going to move `_create_prompt` to utils. that will work better I think.

Reply:
okay that sounds fine!

Reply:
actually, would that be problematic if I made `_create_prompt` public?

Reply:
not necessarily, but I still would want to understand where it's going to be used outside of the class

Reply:
Im thinking about calling it inside `embed_task` in `utils` so that the prompt string embedded is the same as the string used in prompting the model.

Reply:
got it, I thought that you were considering moving `embed_task` inside of the approach. I think both `embed_task` and `create_prompt` are kinda the same category of thing, so I would recommend either making them both utilities, or both approach methods.

Reply:
sounds good. The problem was that some other functions rely on it as well, Ill just move all the functions related to the embedding and cosine sim into approach. Ill also just move the test cases I have into the correct new location.

Reply:
how should I access `_create_prompt` if I want to use it in a test case?

Reply:
You can disable protected-access for the sake of unit tests. Search the code for "protected-access" to see an example